File: src/main/java/net/zargor/afterlife/web/WebRequest.java
package net.zargor.afterlife.web;

/**
 * Needed by the [HttpHandler]. Annotation defines the route
 */
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
public @interface WebRequest {

    String route();

    boolean needToLogged();

    GroupPermissions[] groupNeededRights();
}

File: src/main/kotlin/net/zargor/afterlife/web/WebRequest.kt
package net.zargor.afterlife.web

import net.zargor.afterlife.web.objects.GroupPermissions

/**
 * Needed by the [HttpHandler]. Annotation defines the route
 */
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
annotation class WebRequest(val route: String, val permission: GroupPermissions)